### ISSUE TITLE
[java] ImmutableField: False positive when variable is updated in conditional loop

### ISSUE THREAD

**Issue by Fernal73** [pmd/pmd#2125]
**Affects PMD Version:** 
6.18.0, 6.19.0

**Rule:**
[ImmutableField](https://pmd.github.io/latest/pmd_rules_java_design.html#immutablefield)

**Description:**
If field is updated in conditional loop, this rule still identifies the field as a possible final or immutable candidate.

**Code Sample demonstrating the issue:**

``` java
package pmdtests;

import java.util.Random;
import java.math.BigInteger;

public class TestFinal {
  private BigInteger e;

  public TestFinal() {
    Random random = new Random();
    e = BigInteger.probablePrime(Integer.MAX_VALUE / 2, random);

    while (random.nextBoolean()) 
      e = e.add(BigInteger.ONE);
  }

}

```

**Running PMD through:** *[CLI | Ant **

This is sample code. Don't expect it to do anything useful.
